### ISSUE TITLE
System.ArgumentException: The path is not of a legal form.

### ISSUE THREAD

**Issue by borismod** [gitextensions/gitextensions#3186]
i get System.ArgumentException: The path is not of a legal form. when I open GitExtensions and it closes.

```
System.String NormalizePath(System.String, Boolean, Int32, Boolean) @ System.IO.Path

 at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.GetFullPathInternal(String path)
   at System.IO.Path.GetFullPath(String path)
   at System.IO.FileSystemWatcher.StartRaisingEvents()
   at GitCommands.Settings.FileSettingsCache.LoadImpl()
   at GitCommands.SettingsCache.<>c__DisplayClass1.<LockedAction>b__0()
   at GitCommands.SettingsCache.LockedAction[T](Func`1 action)
   at GitCommands.SettingsCache.<>c__DisplayClass1.<LockedAction>b__0()
   at GitCommands.SettingsCache.LockedAction[T](Func`1 action)
   at GitCommands.SettingsCache.EnsureSettingsAreUpToDate()
   at GitCommands.SettingsCache.<>c__DisplayClass15`1.<TryGetValue>b__14()
   at GitCommands.SettingsCache.LockedAction[T](Func`1 action)
   at GitCommands.SettingsCache.TryGetValue[T](String name, T defaultValue, Func`2 decode, T& value)
   at GitCommands.Settings.SettingsContainer`2.TryGetValue[T](String name, T defaultValue, Func`2 decode, T& value)
   at GitCommands.Settings.SettingsContainer`2.GetValue[T](String name, T defaultValue, Func`2 decode)
   at GitCommands.Settings.ConfigFileSettings.GetEncoding(String settingName)
   at GitUI.CommandsDialogs.SettingsDialog.Pages.GitConfigSettingsPage.SettingsToPage()
   at GitUI.CommandsDialogs.SettingsDialog.SettingsPageBase.LoadSettings()
   at GitUI.CommandsDialogs.FormSettings.LoadSettings()
   at GitUI.CommandsDialogs.FormSettings.FormSettings_Shown(Object sender, EventArgs e)
   at System.Windows.Forms.Form.OnShown(EventArgs e)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbacks()
```

![image](https://cloud.githubusercontent.com/assets/209971/14816763/6671763a-0bbb-11e6-85fe-031204af681b.png)

**Comment by loneshark99** [comment]
@KindDragon I can try to fix this bug, what would be the repro steps for this.

**Comment by MojaveWastelander** [comment]
It is the same problem as specified in this [issue](https://github.com/gitextensions/gitextensions/issues/2886). In my case "config" file contain putty key path and github linked repositories. What's interesting is that this exception does not appears debugging is done step by step when exception should appear. Here is how I reproduced it:
1) Create an empty "config" file in exe directory
2) Comment following [catch block](https://github.com/gitextensions/gitextensions/blob/master/GitUI/CommandsDialogs/FormSettings.cs#L196):

>         public void LoadSettings()
>         {
>             //try
>             //{
>                 foreach (var settingsPage in SettingsPages)
>                 {
>                     settingsPage.LoadSettings();
>                 }
>             //}
>             //catch (Exception e)
>             //{
>             //    ExceptionUtils.ShowException(e);
>             //    // Bail out before the user saves the incompletely loaded settings
>             //    // and has their day ruined.
>             //    DialogResult = DialogResult.Abort;
>             //
>             //    throw;
>             //}
>         } 

3) Run in debug mode and exception will be thrown in GitCommands.dll!GitCommands.Settings.FileSettingsCache.LoadImpl() Line 166

I observed that the bug appears when I use exe from another machine (like using GitExt in portable mode).

**Comment by jbialobr** [comment]
Fixed in 01c252aa7edf27b508a77c074a7839c3d713b155
